src/context/AlertContext.tsx

```diff
@@ -0,0 +1,11 @@
+import { createContext } from 'react';
+
+type AlertContextValues = {
+  showAlert: boolean;
+  setShowAlert: Dispatch<SetStateAction<boolean>>;
+};
+
+export const AlertContext: Context<AlertContextValues> = createContext<AlertContextValues>({
+  showAlert: false,
+  setShowAlert: () => {},
+});
```

Comment:
lets move this here too:

```
export const AlertProvider = ({ children }: Props) => {
  return (
    <AlertContext.Provider value={{ showAlert, setShowAlert }}>
      {children}
    </AlertContext.Provider>
    )
}
export const useAlert = () => {
  const context = useContext(CartContext);
  if (!context) throw new Error('useAlert must be used within AlertProvider');

  return context;
};
```

then in page.tsx:
```
<AlertProvider>
  ...
</AlertProvider>
```

in child components:
```
  const { setShowAlert } = useAlert();
```

this looks more easier to read and less imports as well

Reply:
already tried in here https://github.com/RR-Ardhika/murojaah/pull/7/commits/7338b46ed5d5f16e7fa0774c6ea19a1df9d3260f

but got error useAlert must be used within AlertProvider

Reply:
useAlert must not be declared in the same component as <AlertProvider />, should be a child component

Reply:
tested after fixed the typo and the useAlert worked even if we declare it in the same component

updated in https://github.com/RR-Ardhika/murojaah/commit/a4ba87dbc135213428b4000a6f047d111d4a6f29, thank you

Reply:
but we need to create new component here https://github.com/RR-Ardhika/murojaah/pull/7/commits/103d175f25024f1016aad234428eda55da244269

because we call to useAlert before using AlertProvider